src/main/kotlin/org/rust/ide/newProject/state/RsUserTemplatesStateImpl.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.newProject.state
+
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import com.intellij.util.xmlb.XmlSerializerUtil
+
+@State(name = "RsUserTemplatesState", storages = [Storage("rust.usertemplates.xml")])
+class RsUserTemplatesStateImpl : RsUserTemplatesState(), PersistentStateComponent<RsUserTemplatesStateImpl> {
```

Comment:
Is interface/implementation separation necessary? BTW the [documentation](https://jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_services.html) might be helpful